app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.js

```diff
@@ -369,8 +369,11 @@ angular
         PipelineConfigService.savePipeline(toSave)
           .then(() => $scope.application.pipelineConfigs.refresh(true))
           .then(
-            () => {
-              setOriginal(toSave);
+            (pipelines) => {
+              const latestFromServer = pipelines.find((p) => p.id === toSave.id);
+              const latest = latestFromServer || toSave;
+              setOriginal(latest);
+              this.revertPipelineChanges();
```

Comment:
this might seem odd, but there is an async callback that reverts $scope.pipeline which makes a simple assignment  here get reverted.

Reply:
Looks good, but can you drop a comment here referencing that async callback?

Reply:
since this already merged, I'll just leave this comment here:

https://github.com/christopherthielen/deck/blob/ceb7a4f2cafb06ccf67cc9ad8ebe9e0be31f884f/app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.js#L257-L268

This should be irrelevant if we ever convert these pipeline codes to react

Reply:
Good enough for me!

mergify was too fast.